public/react-services/vis-factory-handler.js

```diff
@@ -114,7 +128,18 @@ export class VisFactoryHandler {
         store.dispatch(updateVis({ update: true }));
         return;
       } catch (error) {
-        return Promise.reject(error);
+        const options = {
+          context: `${VisFactoryHandler.name}.buildAgentsVisualizations`,
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.CRITICAL,
+          store: true,
+          error: {
+            error: error,
+            message: error.message || error,
+            title: error.name || error,
+          },
+        };
+        getErrorOrchestrator().handleError(options);
```

Comment:
question: The error was propagated before the change. Should it throw the error with the changes?